src/vm/runners/cairo_runner.zig

```diff
@@ -169,17 +171,78 @@ pub const CairoRunner = struct {
         return self.runner_mode == .proof_mode_canonical or self.runner_mode == .proof_mode_cairo1;
     }
 
-    pub fn initBuiltins(self: *Self, vm: *CairoVM) !void {
-        vm.builtin_runners = try CairoLayout.setUpBuiltinRunners(
-            self.layout,
-            self.allocator,
-            self.isProofMode(),
-            self.program.builtins,
-        );
+    pub fn initBuiltins(self: *Self, allow_missing_builtins: bool) !void {
+        var program_builtins = std.AutoHashMap(BuiltinName, void).init(self.allocator);
```

Comment:
Here for my part I didn't understand it like that. For me the objective of the hashset of the Rust implementation here was to retrieve the builtins present in the program whereas here you have an empty hashset that doesn't take into account the builtins declared in the `Program`.

For me here you should declare a hashmap which collects all the builtins present in `self.program.builtins` and you should then remove them from the hashmap in each if condition.

What do you think @lana-shanghai?